### PR TITLE
Convert tasks.js to yargs

### PR DESCRIPTION
**Pull request by ace-n**

### CODE REVIEW

File: datastore/tasks.js [hunk 1]
```diff
@@ -175,78 +172,68 @@ function formatTasks (tasks) {
...
+    markDone(options.taskId, function (err) {
+      if (err) {
+        throw err;
+      }
 
-      break;
-    }
-    case 'delete': {
-      taskId = parseInt(input, 10);
+      console.log('Task %d updated successfully.', options.taskId);
```
Comment (ace-n) [inline datastore/tasks.js]: Should this logic go in the parent function (e.g. `markDone`) as it did in the BigQuery samples, or in the `yargs` handler?

Comment (jmdobry) [inline datastore/tasks.js]: Probably should go in the `markDone` function.

File: datastore/tasks.js [hunk 2]
```diff
@@ -149,37 +158,24 @@ function deleteTask (taskId, callback) {
     taskId
   ]);
 
-  datastore.delete(taskKey, callback);
-}
-// [END delete_entity]
-
-// [START format_results]
```
Comment (jmdobry) [inline datastore/tasks.js]: Are the docs going to continue to work with this region tag deleted?

Comment (ace-n) [inline datastore/tasks.js]: They should - this tag didn't come up anywhere when I did a code search.

**Comment by ace-n** [comment]
@jmdobry should be good to review.

File: datastore/tasks.js [hunk 3]
```diff
@@ -152,101 +158,52 @@ function deleteTask (taskId, callback) {
     taskId
   ]);
 
-  datastore.delete(taskKey, callback);
+  datastore.delete(taskKey, function (err) {
+    if (err) {
+      return callback(err);
+    }
+
+    return callback();
```
Comment (jmdobry) [inline datastore/tasks.js]: Add the following above this line: `console.log('Task %d deleted successfully.', taskId);`

**Comment by ace-n** [comment]
Aside: this sample needs tests - but I'm assuming those will go in a separate PR.

**Pull request merged**